docs/operators/csvformatter.md

```diff
@@ -0,0 +1,135 @@
+CsvFormatter
+============
+
+## Operator Objective
+This operator receives a POJO ([Plain Old Java Object](https://en.wikipedia.org/wiki/Plain_Old_Java_Object)) as an incoming tuple, evaluates the fields based on the constraints defined in the schema and converts the data in 
+the incoming POJO to a custom delimited string and finally emits the delimited string.
+
+CsvFormatter supports schema definition as a JSON string. Couple of ways to define schema are as below:
+
+1. **JSON File Based** - Read the file having content stored in JSON format and use that to convert to a string. This can be done using SchemaUtils implementation. Mostly used and recommended approach.
+2. **String Based** - Define the schema as a String in JSON format.
+
+CsvFormatter does not hold any state and is **idempotent**, **fault-tolerant** and **statically/dynamically partitionable**.
+
+## Operator Information
+1. Operator location: ***malhar-contrib***
+2. Available since: ***3.2.0***
+3. Operator state: ***Evolving***
+3. Java Packages:
+    * Operator: ***[com.datatorrent.contrib.formatter.CsvFormatter](https://www.datatorrent.com/docs/apidocs/com/datatorrent/contrib/formatter/CsvFormatter.html)***
+    
+## Properties, Attributes and Ports
+### <a name="props"></a>Properties of POJOEnricher
+| **Property** | **Description** | **Type** | **Mandatory** | **Default Value** |
+| -------- | ----------- | ---- | ------------------ | ------------- |
+| *schema* | Contents of the schema.Schema is specified in a json format. | String | Yes | N/A |
+
+
+### Platform Attributes that influences operator behavior
+| **Attribute** | **Description** | **Type** | **Mandatory** |
+| -------- | ----------- | ---- | ------------------ |
+| *in.TUPLE_CLASS* | TUPLE_CLASS attribute on input port which tells operator the class of POJO which will be incoming | Class or FQCN| Yes |
+
+
+### Ports
+| **Port** | **Description** | **Type** | **Mandatory** |
+| -------- | ----------- | ---- | ------------------ |
+| *in* | Tuples which need to be formatted are received on this port | Object (POJO) | Yes |
+| *out* | Tuples that are formatted are emitted from this port | String | Yes |
+| *err* | Tuples that could not be converted are emitted on this port | String | No |
+
+## Limitations
+Current CsvFormatter contain following limitations:
+
+1. The field names in schema and the pojo field names should match.For eg. if name of the schema field is "customerName", then POJO should contain a field with the same name. 
+2. Only the constraints for Date (format) is supported.
```

Comment:
Also, the fields will be written to file in the same order as specified in schema.json